doc/content/integrations/pubsub/message-types/_index.md

```diff
@@ -0,0 +1,36 @@
+---
+title: "Message Types"
+description: ""
+---
+
+The Application Server can be configured to publish a message for any of the events listed in the Console. Enabling event messaging also allows you to manually configure a **Sub Topic** for that event. If no Sub Topic is specified, events will be published to the configured **Base Topic**.
+
+<!--more-->
+
+>Sub Topics should be specified for at least **Downlink push** and **Downlink replace** in order to schedule messages by publishing to these topics. If both **Downlink push** and **Downlink replace** are enabled, but have no Sub Topic given, messages published to **Base Topic** may be scheduled as either **Downlink push** OR **Downlink replace**. This is an undesirable race condition.
+
+{{< figure src="topics.png" alt="Topics" >}}
+
+## Message Format
+
+JSON messages sent or received by the Application Server are defined in the [Data Formats]({{< ref "/integrations/data-formats" >}}) section.
+
+## Scheduling Downlinks
+
+You can schedule downlink messages by publishing to a topic. For example, using a `nats-pub` client connected to your NATS server:
+
+```bash
+$ nats-pub push-subtopic '{"end_device_ids":{"device_id":"dev1","application_ids":{"application_id":"app1"}},"downlinks":[{"f_port":15,"frm_payload":"vu8=","priority":"NORMAL"}]}'
```

Comment:
This section is about message types, but mentions NATS instructions here. I think that is confusing. What is the goal of this section, other than saying that it follows the data format used everywhere?